packages/protocol/test/stability/exchange.ts

```diff
@@ -686,6 +762,58 @@ contract('Exchange', (accounts: string[]) => {
             })
           })
 
+          describe('when stableBucket has been capped', async () => {
+            beforeEach(async () => {
+              await mockSortedOracles.setMedianRate(
+                stableToken.address,
+                new BigNumber('2e24').multipliedBy(2000)
+              ) // bump the price by 2000, leaving with 4K CELO per dollar
+            })
+
+            it('has the right price after the cap', async () => {
+              const [tradableGold, mintableStable] = await exchange.getBuyAndSellBuckets(false)
+
+              assertEqualBN(
+                new BigNumber(mintableStable.dividedBy(tradableGold)),
+                new BigNumber('4000')
+              )
+            })
+          })
```

Comment:
Feels like there's an assertion missing here to me. The describe says "when stableBucket has been capped" but the test only assets the ratio, which would hold true even if the cap kicked in or not? Maybe I'm missing something.

Reply:
Ok now looking at both tests I think the intent here is that the stableToken supply is 0 by default at the beginning of the test? I would restructure the specs into something like 3 blocks:

- Test lower bound capping => when stableToken supply < minSupplyForStableBucketCap 
- Test for no capping => when stableToken supply > minSupplyForStableBucketCap and updatedBucketSize < cap 
- Test for upper bound capping => when updatedBucketSize > cap

I think that would make the specs more readable if I understand things.